From 1c0d8a69239ce34a9a60d03247c93c426e66a3a8 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 22 Jun 2022 16:57:44 +0400
Subject: [PATCH 01/17] Update gitignore

---
 .gitignore | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/.gitignore b/.gitignore
index c2896799e..d6475a76d 100644
--- a/.gitignore
+++ b/.gitignore
@@ -51,3 +51,6 @@ stage
 *.*~
 .idea/
 cmake-build-debug/
+.cache/
+build/
+compile_commands.json
-- 
2.45.2


From 97ad9365598d48dbb25ee39a3d78854681037682 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 22 Jun 2022 16:57:44 +0400
Subject: [PATCH 02/17] Disable sponsored messages

---
 Telegram/SourceFiles/data/components/sponsored_messages.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/data/components/sponsored_messages.cpp b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
index c20ec967e..bb3de6794 100644
--- a/Telegram/SourceFiles/data/components/sponsored_messages.cpp
+++ b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
@@ -193,7 +193,7 @@ void SponsoredMessages::inject(
 }
 
 bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
-	return history->peer->isChannel();
+	return false;
 }
 
 void SponsoredMessages::request(not_null<History*> history, Fn<void()> done) {
-- 
2.45.2


From 2b76384cfeb84c67ada3ef6c38b3bc87de3a3236 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Fri, 19 May 2023 17:51:54 +0400
Subject: [PATCH 03/17] Cannot react

---
 Telegram/SourceFiles/history/history_item.cpp | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_item.cpp b/Telegram/SourceFiles/history/history_item.cpp
index 73e0662c0..1ff53e111 100644
--- a/Telegram/SourceFiles/history/history_item.cpp
+++ b/Telegram/SourceFiles/history/history_item.cpp
@@ -2479,14 +2479,7 @@ void HistoryItem::translationDone(LanguageId to, TextWithEntities result) {
 }
 
 bool HistoryItem::canReact() const {
-	if (!isRegular() || isService()) {
-		return false;
-	} else if (const auto media = this->media()) {
-		if (media->call()) {
-			return false;
-		}
-	}
-	return true;
+    return false;
 }
 
 void HistoryItem::toggleReaction(
-- 
2.45.2


From 73f141fcc42d756ec6b021e2c80e4c388c4498ff Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Fri, 19 May 2023 17:52:19 +0400
Subject: [PATCH 04/17] Always empty reaction list

---
 Telegram/SourceFiles/history/history_item.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/history/history_item.cpp b/Telegram/SourceFiles/history/history_item.cpp
index 1ff53e111..11e892a30 100644
--- a/Telegram/SourceFiles/history/history_item.cpp
+++ b/Telegram/SourceFiles/history/history_item.cpp
@@ -2512,7 +2512,7 @@ void HistoryItem::updateReactionsUnknown() {
 
 const std::vector<Data::MessageReaction> &HistoryItem::reactions() const {
 	static const auto kEmpty = std::vector<Data::MessageReaction>();
-	return _reactions ? _reactions->list() : kEmpty;
+	return kEmpty;
 }
 
 bool HistoryItem::reactionsAreTags() const {
-- 
2.45.2


From 42ca2b4937d8ef5a2d2a73331f6432d5e55b0aa4 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 22 May 2023 16:29:10 +0400
Subject: [PATCH 05/17] Disable webpage preview on my messages

---
 Telegram/SourceFiles/apiwrap.cpp                             | 5 ++---
 .../history/view/controls/history_view_webpage_processor.cpp | 3 +--
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/Telegram/SourceFiles/apiwrap.cpp b/Telegram/SourceFiles/apiwrap.cpp
index ba4400684..3a5dfcc59 100644
--- a/Telegram/SourceFiles/apiwrap.cpp
+++ b/Telegram/SourceFiles/apiwrap.cpp
@@ -2140,7 +2140,7 @@ void ApiWrap::saveDraftsToCloud() {
 
 		auto flags = MTPmessages_SaveDraft::Flags(0);
 		auto &textWithTags = cloudDraft->textWithTags;
-		if (cloudDraft->webpage.removed) {
+		if (true) {
 			flags |= MTPmessages_SaveDraft::Flag::f_no_webpage;
 		} else if (!cloudDraft->webpage.url.isEmpty()) {
 			flags |= MTPmessages_SaveDraft::Flag::f_media;
@@ -3743,8 +3743,7 @@ void ApiWrap::sendMessage(MessageToSend &&message) {
 			sendFlags |= MTPmessages_SendMessage::Flag::f_reply_to;
 			mediaFlags |= MTPmessages_SendMedia::Flag::f_reply_to;
 		}
-		const auto ignoreWebPage = message.webPage.removed
-			|| (exactWebPage && !isLast);
+		const auto ignoreWebPage = true;
 		const auto manualWebPage = exactWebPage
 			&& !ignoreWebPage
 			&& (message.webPage.manual || (isLast && !isFirst));
diff --git a/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp b/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
index acf09d845..47419aca7 100644
--- a/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
+++ b/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
@@ -345,8 +345,7 @@ void WebpageProcessor::checkNow(bool force) {
 }
 
 void WebpageProcessor::checkPreview() {
-	const auto previewRestricted = _history->peer
-		&& _history->peer->amRestricted(ChatRestriction::EmbedLinks);
+	const auto previewRestricted = true;
 	if (_parsedLinks.empty()) {
 		_draft.removed = false;
 	}
-- 
2.45.2


From a5114266a7da3291843825f03f1381ff3d65c0e3 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Tue, 23 May 2023 17:08:09 +0400
Subject: [PATCH 06/17] Disable webpage previews on other users messages

---
 Telegram/SourceFiles/history/history_item.cpp | 30 ++-----------------
 1 file changed, 2 insertions(+), 28 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_item.cpp b/Telegram/SourceFiles/history/history_item.cpp
index 11e892a30..f5922204a 100644
--- a/Telegram/SourceFiles/history/history_item.cpp
+++ b/Telegram/SourceFiles/history/history_item.cpp
@@ -289,34 +289,8 @@ std::unique_ptr<Data::Media> HistoryItem::CreateMedia(
 		}, [](const MTPDdocumentEmpty &) -> Result {
 			return nullptr;
 		});
-	}, [&](const MTPDmessageMediaWebPage &media) {
-		using Flag = MediaWebPageFlag;
-		const auto flags = Flag()
-			| (media.is_force_large_media()
-				? Flag::ForceLargeMedia
-				: Flag())
-			| (media.is_force_small_media()
-				? Flag::ForceSmallMedia
-				: Flag())
-			| (media.is_manual() ? Flag::Manual : Flag())
-			| (media.is_safe() ? Flag::Safe : Flag());
-		return media.vwebpage().match([](const MTPDwebPageEmpty &) -> Result {
-			return nullptr;
-		}, [&](const MTPDwebPagePending &webpage) -> Result {
-			return std::make_unique<Data::MediaWebPage>(
-				item,
-				item->history()->owner().processWebpage(webpage),
-				flags);
-		}, [&](const MTPDwebPage &webpage) -> Result {
-			return std::make_unique<Data::MediaWebPage>(
-				item,
-				item->history()->owner().processWebpage(webpage),
-				flags);
-		}, [](const MTPDwebPageNotModified &) -> Result {
-			LOG(("API Error: "
-				"webPageNotModified is unexpected in message media."));
-			return nullptr;
-		});
+	}, [&](const MTPDmessageMediaWebPage &media) -> Result {
+		return nullptr;
 	}, [&](const MTPDmessageMediaGame &media) -> Result {
 		return media.vgame().match([&](const MTPDgame &game) {
 			return std::make_unique<Data::MediaGame>(
-- 
2.45.2


From 04e6022170e1c3ed4d0822faf058cd876ac6b2bd Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 22 May 2023 17:11:53 +0400
Subject: [PATCH 07/17] Disable context menu item N Reacted

---
 Telegram/SourceFiles/history/history_inner_widget.cpp          | 3 +--
 .../SourceFiles/history/view/history_view_context_menu.cpp     | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_inner_widget.cpp b/Telegram/SourceFiles/history/history_inner_widget.cpp
index 46d1a4612..633d72dfb 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.cpp
+++ b/Telegram/SourceFiles/history/history_inner_widget.cpp
@@ -2103,8 +2103,7 @@ void HistoryInner::showContextMenu(QContextMenuEvent *e, bool showFromTouch) {
 		return item;
 	};
 	const auto whoReactedItem = groupLeaderOrSelf(_dragStateItem);
-	const auto hasWhoReactedItem = whoReactedItem
-		&& Api::WhoReactedExists(whoReactedItem, Api::WhoReactedList::All);
+	const auto hasWhoReactedItem = false;
 	const auto clickedReaction = link
 		? link->property(
 			kReactionsCountEmojiProperty).value<Data::ReactionId>()
diff --git a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
index a53ab2028..96bd1818f 100644
--- a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
@@ -1168,8 +1168,7 @@ base::unique_qptr<Ui::PopupMenu> FillContextMenu(
 		: nullptr;
 	const auto hasSelection = !request.selectedItems.empty()
 		|| !request.selectedText.empty();
-	const auto hasWhoReactedItem = item
-		&& Api::WhoReactedExists(item, Api::WhoReactedList::All);
+	const auto hasWhoReactedItem = false;
 
 	auto result = base::make_unique_q<Ui::PopupMenu>(
 		list,
-- 
2.45.2


From c54d11bd2fbe623524faa2101676f367309fd8f2 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 10:25:35 +0400
Subject: [PATCH 08/17] Cannot pin messages; disabled PinnedBar

---
 Telegram/SourceFiles/data/data_peer.cpp         | 2 +-
 Telegram/SourceFiles/history/history_widget.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Telegram/SourceFiles/data/data_peer.cpp b/Telegram/SourceFiles/data/data_peer.cpp
index 064f8d91b..505d636c9 100644
--- a/Telegram/SourceFiles/data/data_peer.cpp
+++ b/Telegram/SourceFiles/data/data_peer.cpp
@@ -1194,7 +1194,7 @@ Data::RestrictionCheckResult PeerData::amRestricted(
 				: Result::Allowed())
 			: (right == ChatRestriction::PinMessages)
 			? ((user->flags() & UserDataFlag::CanPinMessages)
-				? Result::Allowed()
+				? Result::Explicit()
 				: Result::Explicit())
 			: Result::Allowed();
 	} else if (const auto channel = asChannel()) {
diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index a805c5479..8da16ee8e 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -7067,7 +7067,7 @@ void HistoryWidget::checkPinnedBarState() {
 		}
 		return;
 	}
-	if (_pinnedBar || !universalPinnedId) {
+	if (true) {
 		return;
 	}
 
-- 
2.45.2


From 2fe8b8bcc742aedfe79588b7646c79a1cf0393eb Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 10:55:01 +0400
Subject: [PATCH 09/17] Disable record button

---
 Telegram/SourceFiles/history/history_widget.cpp           | 8 +-------
 .../view/controls/history_view_compose_controls.cpp       | 7 +------
 2 files changed, 2 insertions(+), 13 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 8da16ee8e..aedb5e991 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -4854,13 +4854,7 @@ bool HistoryWidget::isSearching() const {
 }
 
 bool HistoryWidget::showRecordButton() const {
-	return Media::Capture::instance()->available()
-		&& !_voiceRecordBar->isListenState()
-		&& !_voiceRecordBar->isRecordingByAnotherBar()
-		&& !HasSendText(_field)
-		&& !_previewDrawPreview
-		&& !readyToForward()
-		&& !_editMsgId;
+	return false;
 }
 
 bool HistoryWidget::showInlineBotCancel() const {
diff --git a/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp b/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
index cc4a91a1f..eb8f4eca9 100644
--- a/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
+++ b/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
@@ -1542,12 +1542,7 @@ void ComposeControls::orderControls() {
 }
 
 bool ComposeControls::showRecordButton() const {
-	return ::Media::Capture::instance()->available()
-		&& !_voiceRecordBar->isListenState()
-		&& !_voiceRecordBar->isRecordingByAnotherBar()
-		&& !HasSendText(_field)
-		&& !readyToForward()
-		&& !isEditingMessage();
+	return false;
 }
 
 void ComposeControls::clearListenState() {
-- 
2.45.2


From ca034f19332664ea9d76b92b498b532ca9b0302b Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 16:45:47 +0400
Subject: [PATCH 10/17] Disable calls

---
 .../history/view/history_view_top_bar_widget.cpp     | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
index 59a94d585..d2db627c4 100644
--- a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
@@ -1153,16 +1153,8 @@ void TopBarWidget::updateControlsVisibility() {
 		&& !isOneColumn
 		&& _controller->canShowThirdSection()
 		&& !_chooseForReportReason);
-	const auto callsEnabled = [&] {
-		if (const auto peer = _activeChat.key.peer()) {
-			if (const auto user = peer->asUser()) {
-				return !user->isSelf()
-					&& !user->isBot()
-					&& !peer->isServiceUser();
-			}
-		}
-		return false;
-	}();
+
+	const auto callsEnabled = false;
 	_call->setVisible(historyMode
 		&& callsEnabled
 		&& !_chooseForReportReason);
-- 
2.45.2


From d22d55db76d560f2affe7be24f728201bded32b3 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Thu, 25 May 2023 14:38:14 +0400
Subject: [PATCH 11/17] Disable MUTE/UNMUTE button

---
 Telegram/SourceFiles/history/history_widget.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index aedb5e991..6321e7089 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -4843,10 +4843,7 @@ bool HistoryWidget::isChoosingTheme() const {
 }
 
 bool HistoryWidget::isMuteUnmute() const {
-	return _peer
-		&& ((_peer->isBroadcast() && !_peer->asChannel()->canPostMessages())
-			|| (_peer->isGigagroup() && !Data::CanSendAnything(_peer))
-			|| _peer->isRepliesChat());
+	return false;
 }
 
 bool HistoryWidget::isSearching() const {
-- 
2.45.2


From 1983ec5f7f41fa710585cf7276737c3463e757cb Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Thu, 28 Sep 2023 09:28:21 +0400
Subject: [PATCH 12/17] Disable stories

---
 Telegram/SourceFiles/data/data_peer.cpp                   | 5 -----
 Telegram/SourceFiles/dialogs/dialogs_row.cpp              | 2 +-
 Telegram/SourceFiles/dialogs/dialogs_widget.cpp           | 7 +------
 Telegram/SourceFiles/info/profile/info_profile_widget.cpp | 4 ----
 4 files changed, 2 insertions(+), 16 deletions(-)

diff --git a/Telegram/SourceFiles/data/data_peer.cpp b/Telegram/SourceFiles/data/data_peer.cpp
index 505d636c9..e2315d44a 100644
--- a/Telegram/SourceFiles/data/data_peer.cpp
+++ b/Telegram/SourceFiles/data/data_peer.cpp
@@ -1369,11 +1369,6 @@ const Data::WallPaper *PeerData::wallPaper() const {
 }
 
 bool PeerData::hasActiveStories() const {
-	if (const auto user = asUser()) {
-		return user->hasActiveStories();
-	} else if (const auto channel = asChannel()) {
-		return channel->hasActiveStories();
-	}
 	return false;
 }
 
diff --git a/Telegram/SourceFiles/dialogs/dialogs_row.cpp b/Telegram/SourceFiles/dialogs/dialogs_row.cpp
index 9ecddf88f..dfbfe7d16 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_row.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_row.cpp
@@ -339,7 +339,7 @@ void Row::PaintCornerBadgeFrame(
 	q.translate(framePadding, framePadding);
 	auto hq = std::optional<PainterHighQualityEnabler>();
 	const auto photoSize = context.st->photoSize;
-	const auto storiesCount = data->storiesCount;
+	const auto storiesCount = 0;
 	if (storiesCount) {
 		hq.emplace(q);
 		const auto line = st::dialogsStoriesFull.lineTwice / 2.;
diff --git a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
index 068c30815..1094a7e39 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
@@ -245,12 +245,7 @@ Widget::Widget(
 	object_ptr<Ui::IconButton>(this, st::dialogsLock))
 , _scroll(this)
 , _scrollToTop(_scroll, st::dialogsToUp)
-, _stories((_layout != Layout::Child)
-	? std::make_unique<Stories::List>(
-		this,
-		st::dialogsStoriesList,
-		_storiesContents.events() | rpl::flatten_latest())
-	: nullptr)
+, _stories(nullptr)
 , _searchTimer([=] { search(); })
 , _singleMessageSearch(&controller->session()) {
 	const auto makeChildListShown = [](PeerId peerId, float64 shown) {
diff --git a/Telegram/SourceFiles/info/profile/info_profile_widget.cpp b/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
index 48b0107ed..cd0bac66d 100644
--- a/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
+++ b/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
@@ -119,10 +119,6 @@ rpl::producer<QString> Widget::title() {
 }
 
 rpl::producer<Dialogs::Stories::Content> Widget::titleStories() {
-	const auto peer = controller()->key().peer();
-	if (peer && !peer->isChat()) {
-		return Dialogs::Stories::LastForPeer(peer);
-	}
 	return nullptr;
 }
 
-- 
2.45.2


From 62c333605fe5421a2f6f0efa78b32e461fe3c999 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 8 Nov 2023 14:21:52 +0400
Subject: [PATCH 13/17] Use sccache; use mold linker

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e2ac34c2..e1da9e594 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,6 +35,10 @@ project(Telegram
 )
 set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT Telegram)
 
+set(CMAKE_C_COMPILER_LAUNCHER "/usr/bin/sccache")
+set(CMAKE_CXX_COMPILER_LAUNCHER "/usr/bin/sccache")
+set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=mold")
+
 get_filename_component(third_party_loc "Telegram/ThirdParty" REALPATH)
 get_filename_component(submodules_loc "Telegram" REALPATH)
 get_filename_component(cmake_helpers_loc "cmake" REALPATH)
-- 
2.45.2


From 8a144b241a597ebfa00efac107f77dffb57efeb9 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 8 Nov 2023 17:22:20 +0400
Subject: [PATCH 14/17] Remove ability to copy monospace text via click

---
 Telegram/SourceFiles/core/ui_integration.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Telegram/SourceFiles/core/ui_integration.cpp b/Telegram/SourceFiles/core/ui_integration.cpp
index f3de162ce..abea4caac 100644
--- a/Telegram/SourceFiles/core/ui_integration.cpp
+++ b/Telegram/SourceFiles/core/ui_integration.cpp
@@ -214,10 +214,6 @@ std::shared_ptr<ClickHandler> UiIntegration::createLinkHandler(
 		}
 	} break;
 
-	case EntityType::Code:
-		return std::make_shared<MonospaceClickHandler>(data.text, data.type);
-	case EntityType::Pre:
-		return std::make_shared<MonospaceClickHandler>(data.text, data.type);
 	case EntityType::Phone:
 		return std::make_shared<PhoneClickHandler>(my->session, data.text);
 	}
-- 
2.45.2


From 631ec2007aec8df1ed43ea7845fa3923a9321be8 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Sun, 19 Nov 2023 12:23:06 +0400
Subject: [PATCH 15/17] Disable background color on replies

---
 Telegram/SourceFiles/ui/chat/chat_style.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/ui/chat/chat_style.h b/Telegram/SourceFiles/ui/chat/chat_style.h
index f7499a2d0..5faa60e3f 100644
--- a/Telegram/SourceFiles/ui/chat/chat_style.h
+++ b/Telegram/SourceFiles/ui/chat/chat_style.h
@@ -35,7 +35,7 @@ inline constexpr auto kColorPatternsCount = Text::kMaxQuoteOutlines;
 inline constexpr auto kColorIndexCount = uint8(1 << 6);
 inline constexpr auto kSimpleColorIndexCount = uint8(7);
 
-inline constexpr auto kDefaultBgOpacity = 0.12;
+inline constexpr auto kDefaultBgOpacity = 0;
 inline constexpr auto kDefaultOutline1Opacity = 0.9;
 inline constexpr auto kDefaultOutline2Opacity = 0.3;
 inline constexpr auto kDefaultOutline3Opacity = 0.6;
-- 
2.45.2


From 4bf122d88a4e634cd702cf0372ee8b769c7736cb Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Sun, 19 Nov 2023 12:56:39 +0400
Subject: [PATCH 16/17] Disable background emoji on replies

---
 Telegram/SourceFiles/history/view/history_view_reply.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Telegram/SourceFiles/history/view/history_view_reply.cpp b/Telegram/SourceFiles/history/view/history_view_reply.cpp
index d2043b027..89e9ec772 100644
--- a/Telegram/SourceFiles/history/view/history_view_reply.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_reply.cpp
@@ -626,9 +626,7 @@ void Reply::paint(
 	const auto &quoteSt = _hasQuoteIcon
 		? st::messageTextStyle.blockquote
 		: st::messageQuoteStyle;
-	const auto backgroundEmoji = backgroundEmojiId
-		? st->backgroundEmojiData(backgroundEmojiId).get()
-		: nullptr;
+	const auto backgroundEmoji = (Ui::BackgroundEmojiData*)nullptr;
 	const auto backgroundEmojiCache = backgroundEmoji
 		? &backgroundEmoji->caches[Ui::BackgroundEmojiData::CacheIndex(
 			selected,
-- 
2.45.2


From c4f1a64a5de9ecb674343bf3957ed6bdd31adcc4 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Tue, 12 Dec 2023 14:28:34 +0400
Subject: [PATCH 17/17] Allow copying text from restricted posts

---
 Telegram/SourceFiles/history/history_inner_widget.cpp     | 4 ++--
 .../SourceFiles/media/view/media_view_overlay_widget.cpp  | 8 +-------
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_inner_widget.cpp b/Telegram/SourceFiles/history/history_inner_widget.cpp
index 633d72dfb..ffdc98947 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.cpp
+++ b/Telegram/SourceFiles/history/history_inner_widget.cpp
@@ -2830,12 +2830,12 @@ void HistoryInner::showContextMenu(QContextMenuEvent *e, bool showFromTouch) {
 }
 
 bool HistoryInner::hasCopyRestriction(HistoryItem *item) const {
-	return !_peer->allowsForwarding() || (item && item->forbidsForward());
+	return false;
 }
 
 bool HistoryInner::hasCopyMediaRestriction(
 		not_null<HistoryItem*> item) const {
-	return hasCopyRestriction(item) || item->forbidsSaving();
+	return false;
 }
 
 bool HistoryInner::showCopyRestriction(HistoryItem *item) {
diff --git a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
index 6d7e8d73b..ffd071f4d 100644
--- a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
+++ b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
@@ -1034,13 +1034,7 @@ QSize OverlayWidget::flipSizeByRotation(QSize size) const {
 }
 
 bool OverlayWidget::hasCopyMediaRestriction(bool skipPremiumCheck) const {
-	if (const auto story = _stories ? _stories->story() : nullptr) {
-		return skipPremiumCheck
-			? !story->canDownloadIfPremium()
-			: !story->canDownloadChecked();
-	}
-	return (_history && !_history->peer->allowsForwarding())
-		|| (_message && _message->forbidsSaving());
+	return false;
 }
 
 bool OverlayWidget::showCopyMediaRestriction(bool skipPRemiumCheck) {
-- 
2.45.2

