From 91993e7d98e936b8a178ba3efaeb09de80ef0ff0 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 22 Jun 2022 16:57:44 +0400
Subject: [PATCH 01/18] Update gitignore

---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 8547c2aa0b..81349c3c8c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -70,3 +70,5 @@ settings.local.json
 
 # Generated changelog page (built by CI, deployed to GitHub Pages)
 /docs/changelog/
+
+build/
-- 
2.53.0


From 4340ef78492e7c19feb62b4331c8a8e52d790059 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 22 Jun 2022 16:57:44 +0400
Subject: [PATCH 02/18] Disable sponsored messages

---
 Telegram/SourceFiles/data/components/sponsored_messages.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Telegram/SourceFiles/data/components/sponsored_messages.cpp b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
index 4e68421fff..8fbdcb6702 100644
--- a/Telegram/SourceFiles/data/components/sponsored_messages.cpp
+++ b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
@@ -239,11 +239,6 @@ void SponsoredMessages::inject(
 }
 
 bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
-	if (history->peer->isChannel()) {
-		return true;
-	} else if (const auto user = history->peer->asUser()) {
-		return user->isBot();
-	}
 	return false;
 }
 
-- 
2.53.0


From ca427b274e4d7a85b413bd96445aecf207d0f6b3 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 1 Sep 2025 10:27:46 +0400
Subject: [PATCH 03/18] Disable sponsored ads on videos

---
 Telegram/SourceFiles/data/components/sponsored_messages.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Telegram/SourceFiles/data/components/sponsored_messages.cpp b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
index 8fbdcb6702..d94a0d27d5 100644
--- a/Telegram/SourceFiles/data/components/sponsored_messages.cpp
+++ b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
@@ -243,8 +243,7 @@ bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
 }
 
 bool SponsoredMessages::canHaveFor(not_null<HistoryItem*> item) const {
-	return item->history()->peer->isBroadcast()
-		&& item->isRegular();
+	return false;
 }
 
 bool SponsoredMessages::isTopBarFor(not_null<History*> history) const {
-- 
2.53.0


From ecdd6cba9d9e9205bdb162c8dddcef88968fb5fd Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Fri, 19 May 2023 17:51:54 +0400
Subject: [PATCH 04/18] Cannot react

---
 Telegram/SourceFiles/history/history_item.cpp | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_item.cpp b/Telegram/SourceFiles/history/history_item.cpp
index 5777ba7b68..3b54b77cd7 100644
--- a/Telegram/SourceFiles/history/history_item.cpp
+++ b/Telegram/SourceFiles/history/history_item.cpp
@@ -3098,16 +3098,7 @@ void HistoryItem::translationDone(LanguageId to, TextWithEntities result) {
 }
 
 bool HistoryItem::canReact() const {
-	if (!isRegular()) {
-		return false;
-	} else if (isService()) {
-		return (_flags & MessageFlag::ReactionsAllowed);
-	} else if (const auto media = this->media()) {
-		if (media->call()) {
-			return (_flags & MessageFlag::ReactionsAllowed);
-		}
-	}
-	return true;
+    return false;
 }
 
 void HistoryItem::addPaidReaction(
@@ -4601,6 +4592,7 @@ void HistoryItem::createComponentsHelper(HistoryItemCommonFields &&fields) {
 }
 
 void HistoryItem::setReactions(const MTPMessageReactions *reactions) {
+    return;
 	Expects(!_reactions);
 
 	if (changeReactions(reactions) && _reactions->hasUnread()) {
@@ -4609,6 +4601,7 @@ void HistoryItem::setReactions(const MTPMessageReactions *reactions) {
 }
 
 void HistoryItem::updateReactions(const MTPMessageReactions *reactions) {
+    return;
 	const auto wasRecentUsers = LookupRecentUnreadReactedUsers(this);
 	const auto hadUnread = hasUnreadReaction();
 	const auto changed = changeReactions(reactions);
-- 
2.53.0


From 0120ffbfdf247f41d8d719bd870ae66e0c0b79ec Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 22 May 2023 16:29:10 +0400
Subject: [PATCH 05/18] Disable webpage preview on my messages

---
 Telegram/SourceFiles/apiwrap.cpp                             | 5 ++---
 .../history/view/controls/history_view_webpage_processor.cpp | 3 +--
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/Telegram/SourceFiles/apiwrap.cpp b/Telegram/SourceFiles/apiwrap.cpp
index ccc2891b19..b52e39e412 100644
--- a/Telegram/SourceFiles/apiwrap.cpp
+++ b/Telegram/SourceFiles/apiwrap.cpp
@@ -2195,7 +2195,7 @@ void ApiWrap::saveDraftsToCloud() {
 
 		auto flags = MTPmessages_SaveDraft::Flags(0);
 		auto &textWithTags = cloudDraft->textWithTags;
-		if (cloudDraft->webpage.removed) {
+		if (true) {
 			flags |= MTPmessages_SaveDraft::Flag::f_no_webpage;
 		} else if (!cloudDraft->webpage.url.isEmpty()) {
 			flags |= MTPmessages_SaveDraft::Flag::f_media;
@@ -4060,8 +4060,7 @@ void ApiWrap::sendMessage(
 			sendFlags |= MTPmessages_SendMessage::Flag::f_reply_to;
 			mediaFlags |= MTPmessages_SendMedia::Flag::f_reply_to;
 		}
-		const auto ignoreWebPage = message.webPage.removed
-			|| (exactWebPage && !isLast);
+		const auto ignoreWebPage = true;
 		const auto manualWebPage = exactWebPage
 			&& !ignoreWebPage
 			&& (message.webPage.manual || (isLast && !isFirst));
diff --git a/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp b/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
index feb0a5edf8..818f4a4696 100644
--- a/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
+++ b/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
@@ -350,8 +350,7 @@ void WebpageProcessor::checkNow(bool force) {
 }
 
 void WebpageProcessor::checkPreview() {
-	const auto previewRestricted = _history->peer
-		&& _history->peer->amRestricted(ChatRestriction::EmbedLinks);
+	const auto previewRestricted = true;
 	if (_parsedLinks.empty()) {
 		_draft.removed = false;
 	}
-- 
2.53.0


From c121132fa18b15c52c94ad204c2f7b9619e331d4 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Tue, 23 May 2023 17:08:09 +0400
Subject: [PATCH 06/18] Disable webpage previews on other users messages

---
 Telegram/SourceFiles/history/history_item.cpp | 30 ++-----------------
 1 file changed, 2 insertions(+), 28 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_item.cpp b/Telegram/SourceFiles/history/history_item.cpp
index 3b54b77cd7..5656ca2db0 100644
--- a/Telegram/SourceFiles/history/history_item.cpp
+++ b/Telegram/SourceFiles/history/history_item.cpp
@@ -322,34 +322,8 @@ std::unique_ptr<Data::Media> HistoryItem::CreateMedia(
 		}, [](const MTPDdocumentEmpty &) -> Result {
 			return nullptr;
 		});
-	}, [&](const MTPDmessageMediaWebPage &media) {
-		using Flag = MediaWebPageFlag;
-		const auto flags = Flag()
-			| (media.is_force_large_media()
-				? Flag::ForceLargeMedia
-				: Flag())
-			| (media.is_force_small_media()
-				? Flag::ForceSmallMedia
-				: Flag())
-			| (media.is_manual() ? Flag::Manual : Flag())
-			| (media.is_safe() ? Flag::Safe : Flag());
-		return media.vwebpage().match([](const MTPDwebPageEmpty &) -> Result {
-			return nullptr;
-		}, [&](const MTPDwebPagePending &webpage) -> Result {
-			return std::make_unique<Data::MediaWebPage>(
-				item,
-				item->history()->owner().processWebpage(webpage),
-				flags);
-		}, [&](const MTPDwebPage &webpage) -> Result {
-			return std::make_unique<Data::MediaWebPage>(
-				item,
-				item->history()->owner().processWebpage(webpage),
-				flags);
-		}, [](const MTPDwebPageNotModified &) -> Result {
-			LOG(("API Error: "
-				"webPageNotModified is unexpected in message media."));
-			return nullptr;
-		});
+	}, [&](const MTPDmessageMediaWebPage &media) -> Result {
+		return nullptr;
 	}, [&](const MTPDmessageMediaGame &media) -> Result {
 		return media.vgame().match([&](const MTPDgame &game) {
 			return std::make_unique<Data::MediaGame>(
-- 
2.53.0


From 05c42d77a6a6c56354583dfd2e0e677204dc31ef Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 22 May 2023 17:11:53 +0400
Subject: [PATCH 07/18] Disable context menu item N Reacted

---
 Telegram/SourceFiles/history/history_inner_widget.cpp          | 3 +--
 .../SourceFiles/history/view/history_view_context_menu.cpp     | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_inner_widget.cpp b/Telegram/SourceFiles/history/history_inner_widget.cpp
index 2ea6998de4..612506c6ac 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.cpp
+++ b/Telegram/SourceFiles/history/history_inner_widget.cpp
@@ -2524,8 +2524,7 @@ void HistoryInner::showContextMenu(QContextMenuEvent *e, bool showFromTouch) {
 		return item;
 	};
 	const auto leaderOrSelf = groupLeaderOrSelf(_dragStateItem);
-	const auto hasWhoReactedItem = leaderOrSelf
-		&& Api::WhoReactedExists(leaderOrSelf, Api::WhoReactedList::All);
+	const auto hasWhoReactedItem = false;
 	using namespace HistoryView::Reactions;
 	const auto clickedReaction = ReactionIdOfLink(link);
 	const auto linkPhoneNumber = link
diff --git a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
index 543d7c6281..b05b203d15 100644
--- a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
@@ -1344,8 +1344,7 @@ void FillContextMenuItems(
 		: nullptr;
 	const auto hasSelection = !request.selectedItems.empty()
 		|| !request.selectedText.empty();
-	const auto hasWhoReactedItem = item
-		&& Api::WhoReactedExists(item, Api::WhoReactedList::All);
+	const auto hasWhoReactedItem = false;
 
 	AddReplyToMessageAction(result, request, list);
 	if (item) {
-- 
2.53.0


From 2b570de17a9eaaa805f4d7d5dce3ac6c47dce758 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 10:25:35 +0400
Subject: [PATCH 08/18] Cannot pin messages; disabled PinnedBar

---
 Telegram/SourceFiles/data/data_peer.cpp         | 2 +-
 Telegram/SourceFiles/history/history_widget.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Telegram/SourceFiles/data/data_peer.cpp b/Telegram/SourceFiles/data/data_peer.cpp
index 96cceb4d49..6530f8c6a1 100644
--- a/Telegram/SourceFiles/data/data_peer.cpp
+++ b/Telegram/SourceFiles/data/data_peer.cpp
@@ -1751,7 +1751,7 @@ Data::RestrictionCheckResult PeerData::amRestricted(
 				: Result::Allowed())
 			: (right == ChatRestriction::PinMessages)
 			? ((user->flags() & UserDataFlag::CanPinMessages)
-				? Result::Allowed()
+				? Result::Explicit()
 				: Result::Explicit())
 			: Result::Allowed();
 	} else if (const auto channel = asChannel()) {
diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 31be1ad830..15d979270a 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -8242,7 +8242,7 @@ void HistoryWidget::checkPinnedBarState() {
 		}
 		return;
 	}
-	if (_pinnedBar || !universalPinnedId) {
+	if (true) {
 		return;
 	}
 
-- 
2.53.0


From bee022e03ba1ed9802a6a6f0304b71ef245b19eb Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 10:55:01 +0400
Subject: [PATCH 09/18] Disable record button

---
 Telegram/SourceFiles/history/history_widget.cpp           | 8 +-------
 .../view/controls/history_view_compose_controls.cpp       | 8 +-------
 2 files changed, 2 insertions(+), 14 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 15d979270a..913d79b92f 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -5768,13 +5768,7 @@ bool HistoryWidget::isSearching() const {
 }
 
 bool HistoryWidget::showRecordButton() const {
-	return (_recordAvailability != Webrtc::RecordAvailability::None)
-		&& !_voiceRecordBar->isListenState()
-		&& !_voiceRecordBar->isRecordingByAnotherBar()
-		&& !HasSendText(_field)
-		&& !_previewDrawPreview
-		&& (_replyTo || !readyToForward())
-		&& !_editMsgId;
+	return false;
 }
 
 bool HistoryWidget::showInlineBotCancel() const {
diff --git a/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp b/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
index dc997eaa6d..39e4fb618a 100644
--- a/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
+++ b/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
@@ -2229,13 +2229,7 @@ void ComposeControls::orderControls() {
 }
 
 bool ComposeControls::showRecordButton() const {
-	return _features.recordMediaMessage
-		&& (_recordAvailability != Webrtc::RecordAvailability::None)
-		&& !_voiceRecordBar->isListenState()
-		&& !_voiceRecordBar->isRecordingByAnotherBar()
-		&& !HasSendText(_field)
-		&& !readyToForward()
-		&& !isEditingMessage();
+	return false;
 }
 
 bool ComposeControls::showEditStarsButton() const {
-- 
2.53.0


From 3d3bc851aca6c22a56fd2f16d2e1184c2579ca3e Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 16:45:47 +0400
Subject: [PATCH 10/18] Disable calls

---
 .../view/history_view_top_bar_widget.cpp      | 25 +++----------------
 1 file changed, 3 insertions(+), 22 deletions(-)

diff --git a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
index 7e92eeeb6e..1d819d74e1 100644
--- a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
@@ -1304,31 +1304,12 @@ void TopBarWidget::updateControlsVisibility() {
 		&& !isOneColumn
 		&& _controller->canShowThirdSection()
 		&& !_chooseForReportReason);
-	const auto callsEnabled = [&] {
-		if (const auto peer = _activeChat.key.peer()) {
-			if (const auto user = peer->asUser()) {
-				return !user->isSelf()
-					&& !user->isBot()
-					&& !user->isInaccessible()
-					&& !peer->isServiceUser();
-			}
-		}
-		return false;
-	}();
+
+	const auto callsEnabled = false;
 	_call->setVisible(historyMode
 		&& callsEnabled
 		&& !_chooseForReportReason);
-	const auto groupCallsEnabled = [&] {
-		if (const auto peer = _activeChat.key.peer()) {
-			if (!peer->isUser() && peer->canManageGroupCall()) {
-				return true;
-			} else if (const auto call = peer->groupCall()) {
-				return (call->fullCount() == 0);
-			}
-			return false;
-		}
-		return false;
-	}();
+	const auto groupCallsEnabled = false;
 	_groupCall->setVisible(historyMode
 		&& groupCallsEnabled
 		&& !_chooseForReportReason);
-- 
2.53.0


From c4fcb486b0ee8d9686d09c14c581753a924c6ddd Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Thu, 25 May 2023 14:38:14 +0400
Subject: [PATCH 11/18] Disable MUTE/UNMUTE button

---
 Telegram/SourceFiles/history/history_widget.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 913d79b92f..be9dfa7acf 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -5756,11 +5756,7 @@ bool HistoryWidget::isChoosingTheme() const {
 }
 
 bool HistoryWidget::isMuteUnmute() const {
-	return _peer
-		&& ((_peer->isBroadcast() && !_peer->asChannel()->canPostMessages())
-			|| (_peer->isGigagroup() && !Data::CanSendAnything(_peer))
-			|| _peer->isRepliesChat()
-			|| _peer->isVerifyCodes());
+	return false;
 }
 
 bool HistoryWidget::isSearching() const {
-- 
2.53.0


From 08be6eaee16318ad0a1ba4e7111eeae25e270969 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Thu, 28 Sep 2023 09:28:21 +0400
Subject: [PATCH 12/18] Disable stories

---
 Telegram/SourceFiles/data/data_peer.cpp                   | 5 -----
 Telegram/SourceFiles/dialogs/dialogs_row.cpp              | 2 +-
 Telegram/SourceFiles/dialogs/dialogs_widget.cpp           | 7 +------
 Telegram/SourceFiles/info/profile/info_profile_widget.cpp | 4 ----
 4 files changed, 2 insertions(+), 16 deletions(-)

diff --git a/Telegram/SourceFiles/data/data_peer.cpp b/Telegram/SourceFiles/data/data_peer.cpp
index 6530f8c6a1..bea01ae81a 100644
--- a/Telegram/SourceFiles/data/data_peer.cpp
+++ b/Telegram/SourceFiles/data/data_peer.cpp
@@ -1983,11 +1983,6 @@ const Data::WallPaper *PeerData::wallPaper() const {
 }
 
 bool PeerData::hasActiveStories() const {
-	if (const auto user = asUser()) {
-		return user->hasActiveStories();
-	} else if (const auto channel = asChannel()) {
-		return channel->hasActiveStories();
-	}
 	return false;
 }
 
diff --git a/Telegram/SourceFiles/dialogs/dialogs_row.cpp b/Telegram/SourceFiles/dialogs/dialogs_row.cpp
index 87ba11d6bb..e9f0f13705 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_row.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_row.cpp
@@ -403,7 +403,7 @@ void Row::PaintCornerBadgeFrame(
 	q.translate(framePadding, framePadding);
 	auto hq = std::optional<PainterHighQualityEnabler>();
 	const auto photoSize = context.st->photoSize;
-	const auto storiesCount = data->storiesCount;
+	const auto storiesCount = 0;
 	if (storiesCount) {
 		hq.emplace(q);
 		const auto line = st::dialogsStoriesFull.lineTwice / 2.;
diff --git a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
index ec971de510..83dadb4e98 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
@@ -367,12 +367,7 @@ Widget::Widget(
 	object_ptr<Ui::IconButton>(this, st::dialogsLock))
 , _scroll(this)
 , _scrollToTop(_scroll, st::dialogsToUp)
-, _stories((_layout != Layout::Child)
-	? std::make_unique<Stories::List>(
-		this,
-		st::dialogsStoriesList,
-		_storiesContents.events() | rpl::flatten_latest())
-	: nullptr)
+, _stories(nullptr)
 , _searchTimer([=] { search(); })
 , _peerSearch(&controller->session(), Api::PeerSearch::Type::WithSponsored)
 , _singleMessageSearch(&controller->session()) {
diff --git a/Telegram/SourceFiles/info/profile/info_profile_widget.cpp b/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
index d9d24bed01..a44c04e696 100644
--- a/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
+++ b/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
@@ -207,10 +207,6 @@ rpl::producer<QString> Widget::title() {
 }
 
 rpl::producer<Dialogs::Stories::Content> Widget::titleStories() {
-	const auto peer = controller()->key().peer();
-	if (peer && !peer->isChat()) {
-		return Dialogs::Stories::LastForPeer(peer);
-	}
 	return nullptr;
 }
 
-- 
2.53.0


From bfa3a1754da999d49865bd551dd37d67d5a6053a Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 8 Nov 2023 17:22:20 +0400
Subject: [PATCH 13/18] Remove ability to copy monospace text via click

---
 Telegram/SourceFiles/core/ui_integration.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Telegram/SourceFiles/core/ui_integration.cpp b/Telegram/SourceFiles/core/ui_integration.cpp
index 907ff27a1b..4709d0c91e 100644
--- a/Telegram/SourceFiles/core/ui_integration.cpp
+++ b/Telegram/SourceFiles/core/ui_integration.cpp
@@ -379,10 +379,6 @@ std::shared_ptr<ClickHandler> UiIntegration::createLinkHandler(
 		}
 	} break;
 
-	case EntityType::Code:
-		return std::make_shared<MonospaceClickHandler>(data.text, data.type);
-	case EntityType::Pre:
-		return std::make_shared<MonospaceClickHandler>(data.text, data.type);
 	case EntityType::Phone:
 		return (my && my->session)
 			? std::make_shared<PhoneClickHandler>(my->session, data.text)
-- 
2.53.0


From e4f5724117d5c34a7eb81600270e7ac231a4e199 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Sun, 19 Nov 2023 12:23:06 +0400
Subject: [PATCH 14/18] Disable background color on replies

---
 Telegram/SourceFiles/ui/chat/chat_style.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/ui/chat/chat_style.h b/Telegram/SourceFiles/ui/chat/chat_style.h
index 03b5fc9f00..7752ff62a6 100644
--- a/Telegram/SourceFiles/ui/chat/chat_style.h
+++ b/Telegram/SourceFiles/ui/chat/chat_style.h
@@ -36,7 +36,7 @@ inline constexpr auto kColorPatternsCount = Text::kMaxQuoteOutlines;
 inline constexpr auto kColorIndexCount = uint8(1 << 6);
 inline constexpr auto kSimpleColorIndexCount = uint8(7);
 
-inline constexpr auto kDefaultBgOpacity = 0.12;
+inline constexpr auto kDefaultBgOpacity = 0;
 inline constexpr auto kDefaultOutline1Opacity = 0.9;
 inline constexpr auto kDefaultOutline2Opacity = 0.3;
 inline constexpr auto kDefaultOutline3Opacity = 0.6;
-- 
2.53.0


From cc0c5a470e4a43ec75484b0e91cfb4a8345adf5e Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Sun, 19 Nov 2023 12:56:39 +0400
Subject: [PATCH 15/18] Disable background emoji on replies

---
 Telegram/SourceFiles/history/view/history_view_reply.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Telegram/SourceFiles/history/view/history_view_reply.cpp b/Telegram/SourceFiles/history/view/history_view_reply.cpp
index 859b6147ba..916da01c01 100644
--- a/Telegram/SourceFiles/history/view/history_view_reply.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_reply.cpp
@@ -827,9 +827,7 @@ void Reply::paint(
 	const auto &quoteSt = _hasQuoteIcon
 		? st::messageTextStyle.blockquote
 		: st::messageQuoteStyle;
-	const auto backgroundEmojiData = backgroundEmojiId
-		? st->backgroundEmojiData(backgroundEmojiId, colorCollectible).get()
-		: nullptr;
+	const auto backgroundEmojiData = (Ui::BackgroundEmojiData*)nullptr;
 	const auto backgroundEmojiCache = !backgroundEmojiData
 		? nullptr
 		: useColorCollectible
-- 
2.53.0


From b92656967f0160632f3fed12f053a42f62814744 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Tue, 12 Dec 2023 14:28:34 +0400
Subject: [PATCH 16/18] Allow copying text from restricted posts

---
 .../history/history_inner_widget.cpp          |  4 +--
 .../history/view/history_view_list_widget.cpp | 28 +++----------------
 .../media/view/media_view_overlay_widget.cpp  | 11 +-------
 3 files changed, 7 insertions(+), 36 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_inner_widget.cpp b/Telegram/SourceFiles/history/history_inner_widget.cpp
index 612506c6ac..1273d2464a 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.cpp
+++ b/Telegram/SourceFiles/history/history_inner_widget.cpp
@@ -3472,12 +3472,12 @@ void HistoryInner::showContextMenu(QContextMenuEvent *e, bool showFromTouch) {
 }
 
 bool HistoryInner::hasCopyRestriction(HistoryItem *item) const {
-	return !_peer->allowsForwarding() || (item && item->forbidsForward());
+	return false;
 }
 
 bool HistoryInner::hasCopyMediaRestriction(
 		not_null<HistoryItem*> item) const {
-	return hasCopyRestriction(item) || item->forbidsSaving();
+	return false;
 }
 
 bool HistoryInner::showCopyRestriction(HistoryItem *item) {
diff --git a/Telegram/SourceFiles/history/view/history_view_list_widget.cpp b/Telegram/SourceFiles/history/view/history_view_list_widget.cpp
index 2055c8396d..ad5047e778 100644
--- a/Telegram/SourceFiles/history/view/history_view_list_widget.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_list_widget.cpp
@@ -1610,39 +1610,19 @@ bool ListWidget::isEmpty() const {
 }
 
 bool ListWidget::hasCopyRestriction(HistoryItem *item) const {
-	return _delegate->listCopyRestrictionType(item)
-		!= CopyRestrictionType::None;
+	return false;
 }
 
 bool ListWidget::hasCopyMediaRestriction(not_null<HistoryItem*> item) const {
-	return _delegate->listCopyMediaRestrictionType(item)
-		!= CopyRestrictionType::None;
+	return false;
 }
 
 bool ListWidget::showCopyRestriction(HistoryItem *item) {
-	const auto type = _delegate->listCopyRestrictionType(item);
-	if (type == CopyRestrictionType::None) {
-		return false;
-	}
-	_delegate->listUiShow()->showToast((type == CopyRestrictionType::Channel)
-		? tr::lng_error_nocopy_channel(tr::now)
-		: (type == CopyRestrictionType::User)
-		? tr::lng_error_nocopy_user(tr::now)
-		: tr::lng_error_nocopy_group(tr::now));
-	return true;
+    return false;
 }
 
 bool ListWidget::showCopyMediaRestriction(not_null<HistoryItem*> item) {
-	const auto type = _delegate->listCopyMediaRestrictionType(item);
-	if (type == CopyRestrictionType::None) {
-		return false;
-	}
-	_delegate->listUiShow()->showToast((type == CopyRestrictionType::Channel)
-		? tr::lng_error_nocopy_channel(tr::now)
-		: (type == CopyRestrictionType::User)
-		? tr::lng_error_nocopy_user(tr::now)
-		: tr::lng_error_nocopy_group(tr::now));
-	return true;
+    return false;
 }
 
 bool ListWidget::hasCopyRestrictionForSelected() const {
diff --git a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
index c8f073f6e1..e337e2395e 100644
--- a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
+++ b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
@@ -1187,16 +1187,7 @@ QSize OverlayWidget::flipSizeByRotation(QSize size) const {
 }
 
 bool OverlayWidget::hasCopyMediaRestriction(bool skipPremiumCheck) const {
-	if (const auto story = _stories ? _stories->story() : nullptr) {
-		if (story->call()) {
-			return true;
-		}
-		return skipPremiumCheck
-			? !story->canDownloadIfPremium()
-			: !story->canDownloadChecked();
-	}
-	return (_history && !_history->peer->allowsForwarding())
-		|| (_message && _message->forbidsSaving());
+	return false;
 }
 
 bool OverlayWidget::showCopyMediaRestriction(bool skipPRemiumCheck) {
-- 
2.53.0


From 4176ffa16dc2d24f5acc4013be4e3fe179e69dbd Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 4 Jun 2025 10:13:25 +0400
Subject: [PATCH 17/18] Remove top bar suggestions with premium ads and
 birthdays

---
 Telegram/SourceFiles/dialogs/dialogs_widget.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
index 83dadb4e98..9465a3f5bd 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
@@ -718,7 +718,6 @@ Widget::Widget(
 	}
 
 	setupFrozenAccountBar();
-	setupTopBarSuggestions();
 }
 
 void Widget::setupSwipeBack() {
-- 
2.53.0


From 78a2c48a376a494ded3bd8ccdb15d6f8bf4a372f Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 6 Apr 2026 17:45:35 +0400
Subject: [PATCH 18/18] Remove ai button

---
 Telegram/SourceFiles/history/history_widget.cpp | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index be9dfa7acf..758b666a9b 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -6238,20 +6238,7 @@ bool HistoryWidget::fieldOrDisabledShown() const {
 }
 
 bool HistoryWidget::hasEnoughLinesForAi() const {
-	if (!_history
-		|| _voiceRecordBar->isActive()
-		|| session().appConfig().aiComposeStyles().empty()) {
-		return false;
-	}
-	const auto &style = _field->st().style;
-	const auto lineHeight = style.lineHeight
-		? style.lineHeight
-		: style.font->height;
-	const auto margins = _field->fullTextMargins();
-	const auto contentHeight = _field->height()
-		- margins.top()
-		- margins.bottom();
-	return contentHeight >= (3 * lineHeight);
+    return false;
 }
 
 void HistoryWidget::updateAiButtonVisibility() {
-- 
2.53.0

