From eb7c5cd3c2a65b1737a0dc72540b538ffbc05f39 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 22 Jun 2022 16:57:44 +0400
Subject: [PATCH 01/16] Update gitignore

---
 .gitignore | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/.gitignore b/.gitignore
index f893d4bad..cf3c5b89e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -52,3 +52,6 @@ stage
 *.*~
 .idea/
 cmake-build-debug/
+.cache/
+build/
+compile_commands.json
-- 
2.48.1


From f67c59ddcd937bbc88bdd0eb0b2fcb9f4094ed64 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 22 Jun 2022 16:57:44 +0400
Subject: [PATCH 02/16] Disable sponsored messages

---
 Telegram/SourceFiles/data/components/sponsored_messages.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Telegram/SourceFiles/data/components/sponsored_messages.cpp b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
index 5ae9d8b2c..608cd0b2f 100644
--- a/Telegram/SourceFiles/data/components/sponsored_messages.cpp
+++ b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
@@ -210,11 +210,6 @@ void SponsoredMessages::inject(
 }
 
 bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
-	if (history->peer->isChannel()) {
-		return true;
-	} else if (const auto user = history->peer->asUser()) {
-		return user->isBot();
-	}
 	return false;
 }
 
-- 
2.48.1


From d8699027fbdccfd01d52147b70c57d78a0b3f613 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Fri, 19 May 2023 17:51:54 +0400
Subject: [PATCH 03/16] Cannot react

---
 Telegram/SourceFiles/history/history_item.cpp | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_item.cpp b/Telegram/SourceFiles/history/history_item.cpp
index 8bd1c4f72..dd34194dc 100644
--- a/Telegram/SourceFiles/history/history_item.cpp
+++ b/Telegram/SourceFiles/history/history_item.cpp
@@ -2596,16 +2596,7 @@ void HistoryItem::translationDone(LanguageId to, TextWithEntities result) {
 }
 
 bool HistoryItem::canReact() const {
-	if (!isRegular()) {
-		return false;
-	} else if (isService()) {
-		return (_flags & MessageFlag::ReactionsAllowed);
-	} else if (const auto media = this->media()) {
-		if (media->call()) {
-			return (_flags & MessageFlag::ReactionsAllowed);
-		}
-	}
-	return true;
+    return false;
 }
 
 void HistoryItem::addPaidReaction(
@@ -3982,6 +3973,7 @@ void HistoryItem::createComponentsHelper(HistoryItemCommonFields &&fields) {
 }
 
 void HistoryItem::setReactions(const MTPMessageReactions *reactions) {
+    return;
 	Expects(!_reactions);
 
 	if (changeReactions(reactions) && _reactions->hasUnread()) {
@@ -3990,6 +3982,7 @@ void HistoryItem::setReactions(const MTPMessageReactions *reactions) {
 }
 
 void HistoryItem::updateReactions(const MTPMessageReactions *reactions) {
+    return;
 	const auto wasRecentUsers = LookupRecentUnreadReactedUsers(this);
 	const auto hadUnread = hasUnreadReaction();
 	const auto changed = changeReactions(reactions);
-- 
2.48.1


From 0a462c54b045d648784dea6766354ebc5c833a12 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 22 May 2023 16:29:10 +0400
Subject: [PATCH 04/16] Disable webpage preview on my messages

---
 Telegram/SourceFiles/apiwrap.cpp                             | 5 ++---
 .../history/view/controls/history_view_webpage_processor.cpp | 3 +--
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/Telegram/SourceFiles/apiwrap.cpp b/Telegram/SourceFiles/apiwrap.cpp
index edaea5972..034bfff80 100644
--- a/Telegram/SourceFiles/apiwrap.cpp
+++ b/Telegram/SourceFiles/apiwrap.cpp
@@ -2158,7 +2158,7 @@ void ApiWrap::saveDraftsToCloud() {
 
 		auto flags = MTPmessages_SaveDraft::Flags(0);
 		auto &textWithTags = cloudDraft->textWithTags;
-		if (cloudDraft->webpage.removed) {
+		if (true) {
 			flags |= MTPmessages_SaveDraft::Flag::f_no_webpage;
 		} else if (!cloudDraft->webpage.url.isEmpty()) {
 			flags |= MTPmessages_SaveDraft::Flag::f_media;
@@ -3809,8 +3809,7 @@ void ApiWrap::sendMessage(MessageToSend &&message) {
 			sendFlags |= MTPmessages_SendMessage::Flag::f_reply_to;
 			mediaFlags |= MTPmessages_SendMedia::Flag::f_reply_to;
 		}
-		const auto ignoreWebPage = message.webPage.removed
-			|| (exactWebPage && !isLast);
+		const auto ignoreWebPage = true;
 		const auto manualWebPage = exactWebPage
 			&& !ignoreWebPage
 			&& (message.webPage.manual || (isLast && !isFirst));
diff --git a/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp b/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
index 379ffb92e..dc0fb6e73 100644
--- a/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
+++ b/Telegram/SourceFiles/history/view/controls/history_view_webpage_processor.cpp
@@ -350,8 +350,7 @@ void WebpageProcessor::checkNow(bool force) {
 }
 
 void WebpageProcessor::checkPreview() {
-	const auto previewRestricted = _history->peer
-		&& _history->peer->amRestricted(ChatRestriction::EmbedLinks);
+	const auto previewRestricted = true;
 	if (_parsedLinks.empty()) {
 		_draft.removed = false;
 	}
-- 
2.48.1


From 4f5189cea9ccdceb8d252fe00e2f57e81d545e35 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Tue, 23 May 2023 17:08:09 +0400
Subject: [PATCH 05/16] Disable webpage previews on other users messages

---
 Telegram/SourceFiles/history/history_item.cpp | 30 ++-----------------
 1 file changed, 2 insertions(+), 28 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_item.cpp b/Telegram/SourceFiles/history/history_item.cpp
index dd34194dc..82304491a 100644
--- a/Telegram/SourceFiles/history/history_item.cpp
+++ b/Telegram/SourceFiles/history/history_item.cpp
@@ -319,34 +319,8 @@ std::unique_ptr<Data::Media> HistoryItem::CreateMedia(
 		}, [](const MTPDdocumentEmpty &) -> Result {
 			return nullptr;
 		});
-	}, [&](const MTPDmessageMediaWebPage &media) {
-		using Flag = MediaWebPageFlag;
-		const auto flags = Flag()
-			| (media.is_force_large_media()
-				? Flag::ForceLargeMedia
-				: Flag())
-			| (media.is_force_small_media()
-				? Flag::ForceSmallMedia
-				: Flag())
-			| (media.is_manual() ? Flag::Manual : Flag())
-			| (media.is_safe() ? Flag::Safe : Flag());
-		return media.vwebpage().match([](const MTPDwebPageEmpty &) -> Result {
-			return nullptr;
-		}, [&](const MTPDwebPagePending &webpage) -> Result {
-			return std::make_unique<Data::MediaWebPage>(
-				item,
-				item->history()->owner().processWebpage(webpage),
-				flags);
-		}, [&](const MTPDwebPage &webpage) -> Result {
-			return std::make_unique<Data::MediaWebPage>(
-				item,
-				item->history()->owner().processWebpage(webpage),
-				flags);
-		}, [](const MTPDwebPageNotModified &) -> Result {
-			LOG(("API Error: "
-				"webPageNotModified is unexpected in message media."));
-			return nullptr;
-		});
+	}, [&](const MTPDmessageMediaWebPage &media) -> Result {
+		return nullptr;
 	}, [&](const MTPDmessageMediaGame &media) -> Result {
 		return media.vgame().match([&](const MTPDgame &game) {
 			return std::make_unique<Data::MediaGame>(
-- 
2.48.1


From b4aabd5f763ae4028da221b315fa6ef35647cb04 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Mon, 22 May 2023 17:11:53 +0400
Subject: [PATCH 06/16] Disable context menu item N Reacted

---
 Telegram/SourceFiles/history/history_inner_widget.cpp          | 3 +--
 .../SourceFiles/history/view/history_view_context_menu.cpp     | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_inner_widget.cpp b/Telegram/SourceFiles/history/history_inner_widget.cpp
index c8192f37a..35415d240 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.cpp
+++ b/Telegram/SourceFiles/history/history_inner_widget.cpp
@@ -2188,8 +2188,7 @@ void HistoryInner::showContextMenu(QContextMenuEvent *e, bool showFromTouch) {
 		return item;
 	};
 	const auto leaderOrSelf = groupLeaderOrSelf(_dragStateItem);
-	const auto hasWhoReactedItem = leaderOrSelf
-		&& Api::WhoReactedExists(leaderOrSelf, Api::WhoReactedList::All);
+	const auto hasWhoReactedItem = false;
 	using namespace HistoryView::Reactions;
 	const auto clickedReaction = ReactionIdOfLink(link);
 	const auto linkPhoneNumber = link
diff --git a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
index 678f59ac9..e7d883259 100644
--- a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
@@ -1175,8 +1175,7 @@ base::unique_qptr<Ui::PopupMenu> FillContextMenu(
 		: nullptr;
 	const auto hasSelection = !request.selectedItems.empty()
 		|| !request.selectedText.empty();
-	const auto hasWhoReactedItem = item
-		&& Api::WhoReactedExists(item, Api::WhoReactedList::All);
+	const auto hasWhoReactedItem = false;
 
 	auto result = base::make_unique_q<Ui::PopupMenu>(
 		list,
-- 
2.48.1


From eb2bf836e384bf99c7d13b0933db133ec2b78c72 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 10:25:35 +0400
Subject: [PATCH 07/16] Cannot pin messages; disabled PinnedBar

---
 Telegram/SourceFiles/data/data_peer.cpp         | 2 +-
 Telegram/SourceFiles/history/history_widget.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Telegram/SourceFiles/data/data_peer.cpp b/Telegram/SourceFiles/data/data_peer.cpp
index b26234ee6..30c08b2bc 100644
--- a/Telegram/SourceFiles/data/data_peer.cpp
+++ b/Telegram/SourceFiles/data/data_peer.cpp
@@ -1341,7 +1341,7 @@ Data::RestrictionCheckResult PeerData::amRestricted(
 				: Result::Allowed())
 			: (right == ChatRestriction::PinMessages)
 			? ((user->flags() & UserDataFlag::CanPinMessages)
-				? Result::Allowed()
+				? Result::Explicit()
 				: Result::Explicit())
 			: Result::Allowed();
 	} else if (const auto channel = asChannel()) {
diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 88060ecf4..074a5a08a 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -7377,7 +7377,7 @@ void HistoryWidget::checkPinnedBarState() {
 		}
 		return;
 	}
-	if (_pinnedBar || !universalPinnedId) {
+	if (true) {
 		return;
 	}
 
-- 
2.48.1


From 0b2c0446658be6ebafc00e7757b6e247c3f2c3db Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 10:55:01 +0400
Subject: [PATCH 08/16] Disable record button

---
 Telegram/SourceFiles/history/history_widget.cpp           | 8 +-------
 .../view/controls/history_view_compose_controls.cpp       | 7 +------
 2 files changed, 2 insertions(+), 13 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 074a5a08a..c53c08404 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -5072,13 +5072,7 @@ bool HistoryWidget::isSearching() const {
 }
 
 bool HistoryWidget::showRecordButton() const {
-	return (_recordAvailability != Webrtc::RecordAvailability::None)
-		&& !_voiceRecordBar->isListenState()
-		&& !_voiceRecordBar->isRecordingByAnotherBar()
-		&& !HasSendText(_field)
-		&& !_previewDrawPreview
-		&& (_replyTo || !readyToForward())
-		&& !_editMsgId;
+	return false;
 }
 
 bool HistoryWidget::showInlineBotCancel() const {
diff --git a/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp b/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
index 1f88adeec..a85f898e0 100644
--- a/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
+++ b/Telegram/SourceFiles/history/view/controls/history_view_compose_controls.cpp
@@ -1520,12 +1520,7 @@ void ComposeControls::orderControls() {
 }
 
 bool ComposeControls::showRecordButton() const {
-	return (_recordAvailability != Webrtc::RecordAvailability::None)
-		&& !_voiceRecordBar->isListenState()
-		&& !_voiceRecordBar->isRecordingByAnotherBar()
-		&& !HasSendText(_field)
-		&& !readyToForward()
-		&& !isEditingMessage();
+	return false;
 }
 
 void ComposeControls::clearListenState() {
-- 
2.48.1


From c1071845f4ed1bbe9d3d7818c4b3b6922ed0fecd Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 24 May 2023 16:45:47 +0400
Subject: [PATCH 09/16] Disable calls

---
 .../view/history_view_top_bar_widget.cpp      | 25 +++----------------
 1 file changed, 3 insertions(+), 22 deletions(-)

diff --git a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
index 2e9ac641b..2c040f365 100644
--- a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
@@ -1173,31 +1173,12 @@ void TopBarWidget::updateControlsVisibility() {
 		&& !isOneColumn
 		&& _controller->canShowThirdSection()
 		&& !_chooseForReportReason);
-	const auto callsEnabled = [&] {
-		if (const auto peer = _activeChat.key.peer()) {
-			if (const auto user = peer->asUser()) {
-				return !user->isSelf()
-					&& !user->isBot()
-					&& !user->isInaccessible()
-					&& !peer->isServiceUser();
-			}
-		}
-		return false;
-	}();
+
+	const auto callsEnabled = false;
 	_call->setVisible(historyMode
 		&& callsEnabled
 		&& !_chooseForReportReason);
-	const auto groupCallsEnabled = [&] {
-		if (const auto peer = _activeChat.key.peer()) {
-			if (peer->canManageGroupCall()) {
-				return true;
-			} else if (const auto call = peer->groupCall()) {
-				return (call->fullCount() == 0);
-			}
-			return false;
-		}
-		return false;
-	}();
+	const auto groupCallsEnabled = false;
 	_groupCall->setVisible(historyMode
 		&& groupCallsEnabled
 		&& !_chooseForReportReason);
-- 
2.48.1


From f1ca2bf240c7631dd404360d4376c8341576b6f5 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Thu, 25 May 2023 14:38:14 +0400
Subject: [PATCH 10/16] Disable MUTE/UNMUTE button

---
 Telegram/SourceFiles/history/history_widget.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index c53c08404..67b591ef3 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -5060,11 +5060,7 @@ bool HistoryWidget::isChoosingTheme() const {
 }
 
 bool HistoryWidget::isMuteUnmute() const {
-	return _peer
-		&& ((_peer->isBroadcast() && !_peer->asChannel()->canPostMessages())
-			|| (_peer->isGigagroup() && !Data::CanSendAnything(_peer))
-			|| _peer->isRepliesChat()
-			|| _peer->isVerifyCodes());
+	return false;
 }
 
 bool HistoryWidget::isSearching() const {
-- 
2.48.1


From 8f3ad1e9003c5bd937ff05f257457d5ea7d996db Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Thu, 28 Sep 2023 09:28:21 +0400
Subject: [PATCH 11/16] Disable stories

---
 Telegram/SourceFiles/data/data_peer.cpp                   | 5 -----
 Telegram/SourceFiles/dialogs/dialogs_row.cpp              | 2 +-
 Telegram/SourceFiles/dialogs/dialogs_widget.cpp           | 7 +------
 Telegram/SourceFiles/info/profile/info_profile_widget.cpp | 4 ----
 4 files changed, 2 insertions(+), 16 deletions(-)

diff --git a/Telegram/SourceFiles/data/data_peer.cpp b/Telegram/SourceFiles/data/data_peer.cpp
index 30c08b2bc..df4d3c00a 100644
--- a/Telegram/SourceFiles/data/data_peer.cpp
+++ b/Telegram/SourceFiles/data/data_peer.cpp
@@ -1519,11 +1519,6 @@ const Data::WallPaper *PeerData::wallPaper() const {
 }
 
 bool PeerData::hasActiveStories() const {
-	if (const auto user = asUser()) {
-		return user->hasActiveStories();
-	} else if (const auto channel = asChannel()) {
-		return channel->hasActiveStories();
-	}
 	return false;
 }
 
diff --git a/Telegram/SourceFiles/dialogs/dialogs_row.cpp b/Telegram/SourceFiles/dialogs/dialogs_row.cpp
index c39133e52..fcfd2eece 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_row.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_row.cpp
@@ -419,7 +419,7 @@ void Row::PaintCornerBadgeFrame(
 	q.translate(framePadding, framePadding);
 	auto hq = std::optional<PainterHighQualityEnabler>();
 	const auto photoSize = context.st->photoSize;
-	const auto storiesCount = data->storiesCount;
+	const auto storiesCount = 0;
 	if (storiesCount) {
 		hq.emplace(q);
 		const auto line = st::dialogsStoriesFull.lineTwice / 2.;
diff --git a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
index 752e431a9..131fbc652 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_widget.cpp
@@ -354,12 +354,7 @@ Widget::Widget(
 	object_ptr<Ui::IconButton>(this, st::dialogsLock))
 , _scroll(this)
 , _scrollToTop(_scroll, st::dialogsToUp)
-, _stories((_layout != Layout::Child)
-	? std::make_unique<Stories::List>(
-		this,
-		st::dialogsStoriesList,
-		_storiesContents.events() | rpl::flatten_latest())
-	: nullptr)
+, _stories(nullptr)
 , _searchTimer([=] { search(); })
 , _singleMessageSearch(&controller->session()) {
 	const auto makeChildListShown = [](PeerId peerId, float64 shown) {
diff --git a/Telegram/SourceFiles/info/profile/info_profile_widget.cpp b/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
index 48b0107ed..cd0bac66d 100644
--- a/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
+++ b/Telegram/SourceFiles/info/profile/info_profile_widget.cpp
@@ -119,10 +119,6 @@ rpl::producer<QString> Widget::title() {
 }
 
 rpl::producer<Dialogs::Stories::Content> Widget::titleStories() {
-	const auto peer = controller()->key().peer();
-	if (peer && !peer->isChat()) {
-		return Dialogs::Stories::LastForPeer(peer);
-	}
 	return nullptr;
 }
 
-- 
2.48.1


From 994fbb6fc516eaac00f5690f554225ae266c4cb1 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 8 Nov 2023 14:21:52 +0400
Subject: [PATCH 12/16] Use sccache; use mold linker

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9393d81d7..509e08646 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,6 +35,10 @@ project(Telegram
 )
 set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT Telegram)
 
+set(CMAKE_C_COMPILER_LAUNCHER "/usr/bin/sccache")
+set(CMAKE_CXX_COMPILER_LAUNCHER "/usr/bin/sccache")
+set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=mold")
+
 get_filename_component(third_party_loc "Telegram/ThirdParty" REALPATH)
 get_filename_component(submodules_loc "Telegram" REALPATH)
 get_filename_component(cmake_helpers_loc "cmake" REALPATH)
-- 
2.48.1


From 45b05b0fb42bb7411ae32b050165362a8ee65d39 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Wed, 8 Nov 2023 17:22:20 +0400
Subject: [PATCH 13/16] Remove ability to copy monospace text via click

---
 Telegram/SourceFiles/core/ui_integration.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Telegram/SourceFiles/core/ui_integration.cpp b/Telegram/SourceFiles/core/ui_integration.cpp
index 40604a6c4..0d34bd867 100644
--- a/Telegram/SourceFiles/core/ui_integration.cpp
+++ b/Telegram/SourceFiles/core/ui_integration.cpp
@@ -213,10 +213,6 @@ std::shared_ptr<ClickHandler> UiIntegration::createLinkHandler(
 		}
 	} break;
 
-	case EntityType::Code:
-		return std::make_shared<MonospaceClickHandler>(data.text, data.type);
-	case EntityType::Pre:
-		return std::make_shared<MonospaceClickHandler>(data.text, data.type);
 	case EntityType::Phone:
 		return std::make_shared<PhoneClickHandler>(my->session, data.text);
 	}
-- 
2.48.1


From 988e4c7c12631f0915a5649a13419d579bd4b579 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Sun, 19 Nov 2023 12:23:06 +0400
Subject: [PATCH 14/16] Disable background color on replies

---
 Telegram/SourceFiles/ui/chat/chat_style.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/ui/chat/chat_style.h b/Telegram/SourceFiles/ui/chat/chat_style.h
index 6247d3879..c79f06dac 100644
--- a/Telegram/SourceFiles/ui/chat/chat_style.h
+++ b/Telegram/SourceFiles/ui/chat/chat_style.h
@@ -36,7 +36,7 @@ inline constexpr auto kColorPatternsCount = Text::kMaxQuoteOutlines;
 inline constexpr auto kColorIndexCount = uint8(1 << 6);
 inline constexpr auto kSimpleColorIndexCount = uint8(7);
 
-inline constexpr auto kDefaultBgOpacity = 0.12;
+inline constexpr auto kDefaultBgOpacity = 0;
 inline constexpr auto kDefaultOutline1Opacity = 0.9;
 inline constexpr auto kDefaultOutline2Opacity = 0.3;
 inline constexpr auto kDefaultOutline3Opacity = 0.6;
-- 
2.48.1


From a80f0bc463cb267398916667b08a50c91f01cf90 Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Sun, 19 Nov 2023 12:56:39 +0400
Subject: [PATCH 15/16] Disable background emoji on replies

---
 Telegram/SourceFiles/history/view/history_view_reply.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Telegram/SourceFiles/history/view/history_view_reply.cpp b/Telegram/SourceFiles/history/view/history_view_reply.cpp
index d2043b027..89e9ec772 100644
--- a/Telegram/SourceFiles/history/view/history_view_reply.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_reply.cpp
@@ -626,9 +626,7 @@ void Reply::paint(
 	const auto &quoteSt = _hasQuoteIcon
 		? st::messageTextStyle.blockquote
 		: st::messageQuoteStyle;
-	const auto backgroundEmoji = backgroundEmojiId
-		? st->backgroundEmojiData(backgroundEmojiId).get()
-		: nullptr;
+	const auto backgroundEmoji = (Ui::BackgroundEmojiData*)nullptr;
 	const auto backgroundEmojiCache = backgroundEmoji
 		? &backgroundEmoji->caches[Ui::BackgroundEmojiData::CacheIndex(
 			selected,
-- 
2.48.1


From 671c209742b9e4dcee9959c7c9f53fb786b2646d Mon Sep 17 00:00:00 2001
From: vehlwn <36479428+vehlwn@users.noreply.github.com>
Date: Tue, 12 Dec 2023 14:28:34 +0400
Subject: [PATCH 16/16] Allow copying text from restricted posts

---
 .../history/history_inner_widget.cpp          |  4 ++--
 .../history/view/history_view_list_widget.cpp | 24 ++++---------------
 .../media/view/media_view_overlay_widget.cpp  |  8 +------
 3 files changed, 7 insertions(+), 29 deletions(-)

diff --git a/Telegram/SourceFiles/history/history_inner_widget.cpp b/Telegram/SourceFiles/history/history_inner_widget.cpp
index 35415d240..aa0133855 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.cpp
+++ b/Telegram/SourceFiles/history/history_inner_widget.cpp
@@ -2977,12 +2977,12 @@ void HistoryInner::showContextMenu(QContextMenuEvent *e, bool showFromTouch) {
 }
 
 bool HistoryInner::hasCopyRestriction(HistoryItem *item) const {
-	return !_peer->allowsForwarding() || (item && item->forbidsForward());
+	return false;
 }
 
 bool HistoryInner::hasCopyMediaRestriction(
 		not_null<HistoryItem*> item) const {
-	return hasCopyRestriction(item) || item->forbidsSaving();
+	return false;
 }
 
 bool HistoryInner::showCopyRestriction(HistoryItem *item) {
diff --git a/Telegram/SourceFiles/history/view/history_view_list_widget.cpp b/Telegram/SourceFiles/history/view/history_view_list_widget.cpp
index 10cea51b4..dc16505e4 100644
--- a/Telegram/SourceFiles/history/view/history_view_list_widget.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_list_widget.cpp
@@ -1581,35 +1581,19 @@ bool ListWidget::isEmpty() const {
 }
 
 bool ListWidget::hasCopyRestriction(HistoryItem *item) const {
-	return _delegate->listCopyRestrictionType(item)
-		!= CopyRestrictionType::None;
+	return false;
 }
 
 bool ListWidget::hasCopyMediaRestriction(not_null<HistoryItem*> item) const {
-	return _delegate->listCopyMediaRestrictionType(item)
-		!= CopyRestrictionType::None;
+	return false;
 }
 
 bool ListWidget::showCopyRestriction(HistoryItem *item) {
-	const auto type = _delegate->listCopyRestrictionType(item);
-	if (type == CopyRestrictionType::None) {
-		return false;
-	}
-	_delegate->listUiShow()->showToast((type == CopyRestrictionType::Channel)
-		? tr::lng_error_nocopy_channel(tr::now)
-		: tr::lng_error_nocopy_group(tr::now));
-	return true;
+    return false;
 }
 
 bool ListWidget::showCopyMediaRestriction(not_null<HistoryItem*> item) {
-	const auto type = _delegate->listCopyMediaRestrictionType(item);
-	if (type == CopyRestrictionType::None) {
-		return false;
-	}
-	_delegate->listUiShow()->showToast((type == CopyRestrictionType::Channel)
-		? tr::lng_error_nocopy_channel(tr::now)
-		: tr::lng_error_nocopy_group(tr::now));
-	return true;
+	return false;
 }
 
 bool ListWidget::hasCopyRestrictionForSelected() const {
diff --git a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
index e752cc846..2b3f2218d 100644
--- a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
+++ b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
@@ -1104,13 +1104,7 @@ QSize OverlayWidget::flipSizeByRotation(QSize size) const {
 }
 
 bool OverlayWidget::hasCopyMediaRestriction(bool skipPremiumCheck) const {
-	if (const auto story = _stories ? _stories->story() : nullptr) {
-		return skipPremiumCheck
-			? !story->canDownloadIfPremium()
-			: !story->canDownloadChecked();
-	}
-	return (_history && !_history->peer->allowsForwarding())
-		|| (_message && _message->forbidsSaving());
+	return false;
 }
 
 bool OverlayWidget::showCopyMediaRestriction(bool skipPRemiumCheck) {
-- 
2.48.1

